cli/src/semgrep/app/scans.py

```diff
@@ -197,6 +156,14 @@ def fetch_and_init_scan_config(self, meta: Dict[str, Any]) -> None:
         self._skipped_match_based_ids = body.get("triage_ignored_match_based_ids") or []
         self.ignore_patterns = body.get("ignored_files") or []
 
+        if logger.isEnabledFor(logging.DEBUG):
+            config = deepcopy(body)
+            try:
+                config["rule_config"] = json.loads(config["rule_config"])
```

Comment:
is `rule_config` always guaranteed to exist in config? Should we use more defensive code?

Reply:
I don't know when/how it wouldn't be present, but it's a good point that the existing error handling doesn't catch `json.loads(None)` so I can change this to `catch Exception` easily enough to handle all the failures.